Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4246,13 +4261,61 @@ namespace ScriptCanvasEditor
             RemoveSystemTickAction(SystemTickActionFlag::CloseNextTabAction);
             CloseNextTab();
         }
+    }
 
-        if (m_systemTickActions == 0)
+    void MainWindow::ClearSaveAttempt()
+    {
+        m_saveAttemptInProgress = false;
+
+        if (!m_systemTickActions)
         {
             AZ::SystemTickBus::Handler::BusDisconnect();
         }
     }
 
+    bool MainWindow::EvaluateSaveAttempt()
+    {
+        const AZ::s64 k_saveAttemptSeconds = 20;
```

Comment:
Style standards: Constants should be [upper camel case](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#constant-names)

```suggestion
        constexpr AZ::s64 SaveAttemptSeconds = 20;
```